tools/api-markdown-documenter/src/LintApiModel.ts

```diff
@@ -156,6 +168,135 @@ function lintApiItem(
 	}
 }
 
+function lintComment(
+	comment: DocComment,
+	associatedItem: ApiDocumentedItem,
+	apiModel: ApiModel,
+	errors: MutableLinterErrors,
+): void {
+	checkTagsUnderTsdocNode(comment.summarySection, associatedItem, apiModel, errors);
+
+	if (comment.deprecatedBlock !== undefined) {
+		checkTagsUnderTsdocNode(comment.deprecatedBlock, associatedItem, apiModel, errors);
+	}
+
+	if (comment.remarksBlock !== undefined) {
+		checkTagsUnderTsdocNode(comment.remarksBlock, associatedItem, apiModel, errors);
+	}
+
+	if (comment.privateRemarks !== undefined) {
+		checkTagsUnderTsdocNode(comment.privateRemarks, associatedItem, apiModel, errors);
+	}
+
+	checkTagsUnderTsdocNodes(comment.params.blocks, associatedItem, apiModel, errors);
+
+	checkTagsUnderTsdocNodes(comment.typeParams.blocks, associatedItem, apiModel, errors);
+
+	checkTagsUnderTsdocNodes(comment.customBlocks, associatedItem, apiModel, errors);
+}
+
+function checkTagsUnderTsdocNode(
+	node: DocNode,
+	associatedItem: ApiDocumentedItem,
+	apiModel: ApiModel,
+	errors: MutableLinterErrors,
+): void {
+	switch (node.kind) {
+		// Nodes under which links cannot occur
+		case DocNodeKind.CodeSpan:
+		case DocNodeKind.BlockTag:
+		case DocNodeKind.EscapedText:
+		case DocNodeKind.FencedCode:
+		case DocNodeKind.HtmlStartTag:
+		case DocNodeKind.HtmlEndTag:
+		case DocNodeKind.PlainText:
+		case DocNodeKind.SoftBreak: {
+			break;
+		}
+		// Nodes with children ("content")
+		case DocNodeKind.Block:
+		case DocNodeKind.ParamBlock: {
+			assert(node instanceof DocBlock, 'Expected a "DocBlock" node.');
+			checkTagsUnderTsdocNode(node.content, associatedItem, apiModel, errors);
+			break;
+		}
+		// Nodes with children ("nodes")
+		case DocNodeKind.Paragraph:
+		case DocNodeKind.Section: {
+			assert(node instanceof DocNodeContainer, 'Expected a "DocNodeContainer" node.');
+			checkTagsUnderTsdocNodes(node.nodes, associatedItem, apiModel, errors);
+			break;
+		}
+		case DocNodeKind.InlineTag: {
+			assert(node instanceof DocInlineTag, 'Expected a "DocInlineTag" node.');
+			// TODO: malformed tag errors
+			break;
+		}
+		case DocNodeKind.LinkTag: {
+			const result = checkLinkTag(node as DocLinkTag, associatedItem, apiModel);
+			if (result !== undefined) {
+				errors.referenceErrors.add(result);
+			}
+			break;
+		}
+		case DocNodeKind.InheritDocTag: {
+			fail(
+				"Encountered an @inheritDoc tag while walking a TSDoc tree. API-Extractor resolves such tags at a higher level, so this is unexpected.",
```

Comment:
I don't understand this failure message. Can you link to the documentation? I'm not sure what it resolving at a "higher level" means.

Reply:
I added a comment pointing to existing notes I left in `lintApiItem`. Hopefully that sufficiently clarifies things.